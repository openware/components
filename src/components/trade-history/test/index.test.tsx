import React from 'react';
import { shallow } from 'enzyme';

import { TradeHistoryWidget, TradeHistoryWidgetProps } from '../index';
import jsonData from '../story/data/data.json';

const defaultProps: TradeHistoryWidgetProps = {
    style: { height: 300 },
    data: jsonData as any,
};

const setup = (props: Partial<TradeHistoryWidgetProps> = {}) => {
    return shallow(<TradeHistoryWidget {...props} {...defaultProps} />);
};

describe('TradeHistoryWidget', () => {
    it('render', () => {
        expect(setup()).toMatchSnapshot();
    });
});
