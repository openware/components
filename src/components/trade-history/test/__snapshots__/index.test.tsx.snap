// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradeHistoryWidget render 1`] = `
<Component
  header="History"
  style={
    Object {
      "height": 300,
    }
  }
>
  <VTable
    columns={
      Array [
        Object {
          "accessor": [Function],
          "align": "left",
          "header": "Time",
        },
        Object {
          "accessor": [Function],
          "align": "right",
          "header": "Amount",
        },
        Object {
          "accessor": [Function],
          "align": "right",
          "header": "Price",
        },
      ]
    }
    data={
      Array [
        Object {
          "amount": 0.5832,
          "created_at": 1599464707,
          "id": 502501345,
          "market": "btcusdt",
          "price": 10126,
          "taker_type": "sell",
          "total": 5905.4832,
        },
        Object {
          "amount": 0.0011,
          "created_at": 1599464702,
          "id": 502501344,
          "market": "btcusdt",
          "price": 10126,
          "taker_type": "sell",
          "total": 11.1386,
        },
        Object {
          "amount": 0.52,
          "created_at": 1599464685,
          "id": 502501341,
          "market": "btcusdt",
          "price": 10124,
          "taker_type": "sell",
          "total": 0.00526448,
        },
        Object {
          "amount": 0.40119948,
          "created_at": 1599464685,
          "id": 502501340,
          "market": "btcusdt",
          "price": 10125,
          "taker_type": "sell",
          "total": 4062.144735,
        },
        Object {
          "amount": 1,
          "created_at": 1599464673,
          "id": 502501339,
          "market": "btcusdt",
          "price": 10125,
          "taker_type": "sell",
          "total": 10125,
        },
        Object {
          "amount": 0.02917813,
          "created_at": 1599464660,
          "id": 502501337,
          "market": "btcusdt",
          "price": 10126,
          "taker_type": "sell",
          "total": 295.45774438,
        },
        Object {
          "amount": 0.11742187,
          "created_at": 1599464660,
          "id": 502501338,
          "market": "btcusdt",
          "price": 10125,
          "taker_type": "sell",
          "total": 1188.89643375,
        },
        Object {
          "amount": 0.005,
          "created_at": 1599464629,
          "id": 502501336,
          "market": "btcusdt",
          "price": 10127,
          "taker_type": "buy",
          "total": 50.635,
        },
        Object {
          "amount": 0.02021,
          "created_at": 1599464532,
          "id": 502501334,
          "market": "btcusdt",
          "price": 10131,
          "taker_type": "sell",
          "total": 204.74751,
        },
        Object {
          "amount": 0.8285,
          "created_at": 1599464509,
          "id": 502501333,
          "market": "btcusdt",
          "price": 10137,
          "taker_type": "sell",
          "total": 8398.5045,
        },
        Object {
          "amount": 0.4419,
          "created_at": 1599464487,
          "id": 502501332,
          "market": "btcusdt",
          "price": 10144,
          "taker_type": "sell",
          "total": 4482.6336,
        },
        Object {
          "amount": 1,
          "created_at": 1599464423,
          "id": 502501330,
          "market": "btcusdt",
          "price": 10137,
          "taker_type": "sell",
          "total": 10137,
        },
        Object {
          "amount": 0.93322719,
          "created_at": 1599464398,
          "id": 502501328,
          "market": "btcusdt",
          "price": 10138,
          "taker_type": "sell",
          "total": 9461.05725222,
        },
        Object {
          "amount": 0.002,
          "created_at": 1599464398,
          "id": 502501327,
          "market": "btcusdt",
          "price": 10138,
          "taker_type": "sell",
          "total": 20.276,
        },
        Object {
          "amount": 0.06477281,
          "created_at": 1599464398,
          "id": 502501329,
          "market": "btcusdt",
          "price": 10137,
          "taker_type": "sell",
          "total": 656.60197497,
        },
        Object {
          "amount": 0.9832,
          "created_at": 1599464386,
          "id": 502501326,
          "market": "btcusdt",
          "price": 10137,
          "taker_type": "sell",
          "total": 9966.6984,
        },
        Object {
          "amount": 0.933,
          "created_at": 1599464335,
          "id": 502501325,
          "market": "btcusdt",
          "price": 10138,
          "taker_type": "sell",
          "total": 9458.754,
        },
        Object {
          "amount": 0.02706,
          "created_at": 1599464163,
          "id": 502501319,
          "market": "btcusdt",
          "price": 10136,
          "taker_type": "sell",
          "total": 274.28016,
        },
        Object {
          "amount": 0.002,
          "created_at": 1599463998,
          "id": 502501315,
          "market": "btcusdt",
          "price": 10154,
          "taker_type": "buy",
          "total": 20.308,
        },
        Object {
          "amount": 0.001,
          "created_at": 1599463980,
          "id": 502501314,
          "market": "btcusdt",
          "price": 10150,
          "taker_type": "buy",
          "total": 10.15,
        },
        Object {
          "amount": 0.10628,
          "created_at": 1599463976,
          "id": 502501313,
          "market": "btcusdt",
          "price": 10144,
          "taker_type": "sell",
          "total": 1078.10432,
        },
        Object {
          "amount": 0.01779,
          "created_at": 1599463976,
          "id": 502501312,
          "market": "btcusdt",
          "price": 10145,
          "taker_type": "sell",
          "total": 180.47955,
        },
        Object {
          "amount": 0.09871,
          "created_at": 1599463976,
          "id": 502501311,
          "market": "btcusdt",
          "price": 10145,
          "taker_type": "sell",
          "total": 1001.41295,
        },
        Object {
          "amount": 0.02037,
          "created_at": 1599463976,
          "id": 502501310,
          "market": "btcusdt",
          "price": 10145,
          "taker_type": "sell",
          "total": 206.65365,
        },
        Object {
          "amount": 0.21177,
          "created_at": 1599463976,
          "id": 502501308,
          "market": "btcusdt",
          "price": 10145,
          "taker_type": "sell",
          "total": 2148.40665,
        },
        Object {
          "amount": 0.23654,
          "created_at": 1599463976,
          "id": 502501309,
          "market": "btcusdt",
          "price": 10145,
          "taker_type": "sell",
          "total": 2399.6983,
        },
        Object {
          "amount": 0.25632,
          "created_at": 1599463976,
          "id": 502501307,
          "market": "btcusdt",
          "price": 10145,
          "taker_type": "sell",
          "total": 2600.3664,
        },
        Object {
          "amount": 0.06808819,
          "created_at": 1599463947,
          "id": 502501306,
          "market": "btcusdt",
          "price": 10144,
          "taker_type": "sell",
          "total": 690.68659936,
        },
        Object {
          "amount": 0.00973181,
          "created_at": 1599463947,
          "id": 502501305,
          "market": "btcusdt",
          "price": 10145,
          "taker_type": "sell",
          "total": 98.72921245,
        },
        Object {
          "amount": 0.0097,
          "created_at": 1599463893,
          "id": 502501303,
          "market": "btcusdt",
          "price": 10141,
          "taker_type": "sell",
          "total": 98.3677,
        },
        Object {
          "amount": 0.10268427,
          "created_at": 1599463880,
          "id": 502501301,
          "market": "btcusdt",
          "price": 10149,
          "taker_type": "buy",
          "total": 1042.14265623,
        },
        Object {
          "amount": 0.38758392,
          "created_at": 1599463880,
          "id": 502501302,
          "market": "btcusdt",
          "price": 10150,
          "taker_type": "buy",
          "total": 3933.976788,
        },
        Object {
          "amount": 0.00973181,
          "created_at": 1599463880,
          "id": 502501300,
          "market": "btcusdt",
          "price": 10149,
          "taker_type": "buy",
          "total": 98.76813969,
        },
        Object {
          "amount": 0.5006,
          "created_at": 1599463850,
          "id": 502501299,
          "market": "btcusdt",
          "price": 10145,
          "taker_type": "sell",
          "total": 5078.587,
        },
        Object {
          "amount": 0.04823971,
          "created_at": 1599463830,
          "id": 502501297,
          "market": "btcusdt",
          "price": 10153,
          "taker_type": "sell",
          "total": 489.77777563,
        },
        Object {
          "amount": 0.00303388,
          "created_at": 1599463830,
          "id": 502501298,
          "market": "btcusdt",
          "price": 10153,
          "taker_type": "sell",
          "total": 30.80298364,
        },
        Object {
          "amount": 0.14996819,
          "created_at": 1599463737,
          "id": 502501293,
          "market": "btcusdt",
          "price": 10143,
          "taker_type": "sell",
          "total": 1521.12735117,
        },
        Object {
          "amount": 0.00973181,
          "created_at": 1599463737,
          "id": 502501292,
          "market": "btcusdt",
          "price": 10144,
          "taker_type": "sell",
          "total": 98.71948064,
        },
        Object {
          "amount": 0.02620418,
          "created_at": 1599463733,
          "id": 502501291,
          "market": "btcusdt",
          "price": 10148,
          "taker_type": "buy",
          "total": 265.92001864,
        },
        Object {
          "amount": 0.07658,
          "created_at": 1599463733,
          "id": 502501290,
          "market": "btcusdt",
          "price": 10148,
          "taker_type": "buy",
          "total": 777.13384,
        },
        Object {
          "amount": 0.02675,
          "created_at": 1599463733,
          "id": 502501288,
          "market": "btcusdt",
          "price": 10148,
          "taker_type": "buy",
          "total": 271.459,
        },
        Object {
          "amount": 0.02675,
          "created_at": 1599463733,
          "id": 502501289,
          "market": "btcusdt",
          "price": 10148,
          "taker_type": "buy",
          "total": 271.459,
        },
        Object {
          "amount": 0.40046,
          "created_at": 1599463732,
          "id": 502501287,
          "market": "btcusdt",
          "price": 10148,
          "taker_type": "buy",
          "total": 4063.86808,
        },
        Object {
          "amount": 0.02464,
          "created_at": 1599463732,
          "id": 502501286,
          "market": "btcusdt",
          "price": 10148,
          "taker_type": "buy",
          "total": 250.04672,
        },
        Object {
          "amount": 0.43885,
          "created_at": 1599463715,
          "id": 502501285,
          "market": "btcusdt",
          "price": 10147,
          "taker_type": "buy",
          "total": 4453.01095,
        },
        Object {
          "amount": 0.06175,
          "created_at": 1599463715,
          "id": 502501284,
          "market": "btcusdt",
          "price": 10147,
          "taker_type": "buy",
          "total": 626.57725,
        },
        Object {
          "amount": 0.29397572,
          "created_at": 1599463618,
          "id": 502501282,
          "market": "btcusdt",
          "price": 10132,
          "taker_type": "buy",
          "total": 2978.56199504,
        },
        Object {
          "amount": 0.36475051,
          "created_at": 1599463618,
          "id": 502501281,
          "market": "btcusdt",
          "price": 10132,
          "taker_type": "buy",
          "total": 3695.65216732,
        },
        Object {
          "amount": 0.01253,
          "created_at": 1599463604,
          "id": 502501277,
          "market": "btcusdt",
          "price": 10130,
          "taker_type": "buy",
          "total": 126.9289,
        },
        Object {
          "amount": 0.05158704,
          "created_at": 1599463325,
          "id": 502501240,
          "market": "btcusdt",
          "price": 10117,
          "taker_type": "buy",
          "total": 521.90608368,
        },
        Object {
          "amount": 0.05158704,
          "created_at": 1599463325,
          "id": 502501239,
          "market": "btcusdt",
          "price": 10117,
          "taker_type": "buy",
          "total": 521.90608368,
        },
        Object {
          "amount": 0.00365,
          "created_at": 1599463174,
          "id": 502501236,
          "market": "btcusdt",
          "price": 10110,
          "taker_type": "buy",
          "total": 36.9015,
        },
        Object {
          "amount": 0.03322,
          "created_at": 1599463174,
          "id": 502501237,
          "market": "btcusdt",
          "price": 10110,
          "taker_type": "buy",
          "total": 335.8542,
        },
        Object {
          "amount": 0.0468,
          "created_at": 1599463030,
          "id": 502501232,
          "market": "btcusdt",
          "price": 10112,
          "taker_type": "sell",
          "total": 473.2416,
        },
        Object {
          "amount": 0.04945985,
          "created_at": 1599463030,
          "id": 502501231,
          "market": "btcusdt",
          "price": 10113,
          "taker_type": "sell",
          "total": 500.18746305,
        },
        Object {
          "amount": 0.0111975,
          "created_at": 1599463028,
          "id": 502501230,
          "market": "btcusdt",
          "price": 10116,
          "taker_type": "sell",
          "total": 113.27391,
        },
        Object {
          "amount": 0.00673193,
          "created_at": 1599463028,
          "id": 502501229,
          "market": "btcusdt",
          "price": 10116,
          "taker_type": "sell",
          "total": 68.10020388,
        },
        Object {
          "amount": 0.5006,
          "created_at": 1599463027,
          "id": 502501228,
          "market": "btcusdt",
          "price": 10121,
          "taker_type": "buy",
          "total": 5066.5726,
        },
        Object {
          "amount": 0.03183272,
          "created_at": 1599463026,
          "id": 502501227,
          "market": "btcusdt",
          "price": 10120,
          "taker_type": "sell",
          "total": 322.1471264,
        },
        Object {
          "amount": 1,
          "created_at": 1599463025,
          "id": 502501226,
          "market": "btcusdt",
          "price": 10120,
          "taker_type": "sell",
          "total": 10120,
        },
        Object {
          "amount": 0.10118592,
          "created_at": 1599463025,
          "id": 502501225,
          "market": "btcusdt",
          "price": 10120,
          "taker_type": "sell",
          "total": 1024.0015104,
        },
        Object {
          "amount": 0.15362,
          "created_at": 1599463025,
          "id": 502501224,
          "market": "btcusdt",
          "price": 10120,
          "taker_type": "sell",
          "total": 1554.6344,
        },
        Object {
          "amount": 0.00969199,
          "created_at": 1599463017,
          "id": 502501223,
          "market": "btcusdt",
          "price": 10122,
          "taker_type": "buy",
          "total": 98.10232278,
        },
        Object {
          "amount": 0.69133091,
          "created_at": 1599463013,
          "id": 502501221,
          "market": "btcusdt",
          "price": 10120,
          "taker_type": "sell",
          "total": 6996.2688092,
        },
        Object {
          "amount": 0.19616909,
          "created_at": 1599463013,
          "id": 502501222,
          "market": "btcusdt",
          "price": 10120,
          "taker_type": "sell",
          "total": 1985.2311908,
        },
        Object {
          "amount": 0.9341,
          "created_at": 1599463001,
          "id": 502501220,
          "market": "btcusdt",
          "price": 10121,
          "taker_type": "sell",
          "total": 9454.0261,
        },
        Object {
          "amount": 0.07825,
          "created_at": 1599462952,
          "id": 502501218,
          "market": "btcusdt",
          "price": 10121,
          "taker_type": "sell",
          "total": 791.96825,
        },
        Object {
          "amount": 0.9855,
          "created_at": 1599462831,
          "id": 502501216,
          "market": "btcusdt",
          "price": 10112,
          "taker_type": "sell",
          "total": 9965.376,
        },
        Object {
          "amount": 0.004,
          "created_at": 1599462819,
          "id": 502501214,
          "market": "btcusdt",
          "price": 10111,
          "taker_type": "sell",
          "total": 40.444,
        },
        Object {
          "amount": 0.004,
          "created_at": 1599462798,
          "id": 502501213,
          "market": "btcusdt",
          "price": 10108,
          "taker_type": "sell",
          "total": 40.432,
        },
        Object {
          "amount": 0.00400801,
          "created_at": 1599462711,
          "id": 502501212,
          "market": "btcusdt",
          "price": 10108,
          "taker_type": "buy",
          "total": 40.51296508,
        },
        Object {
          "amount": 0.4681,
          "created_at": 1599462667,
          "id": 502501210,
          "market": "btcusdt",
          "price": 10109,
          "taker_type": "sell",
          "total": 4732.0229,
        },
        Object {
          "amount": 0.931,
          "created_at": 1599462628,
          "id": 502501209,
          "market": "btcusdt",
          "price": 10112,
          "taker_type": "sell",
          "total": 9414.272,
        },
        Object {
          "amount": 0.11229,
          "created_at": 1599462527,
          "id": 502501200,
          "market": "btcusdt",
          "price": 10117,
          "taker_type": "buy",
          "total": 1136.03793,
        },
        Object {
          "amount": 0.6036,
          "created_at": 1599462507,
          "id": 502501199,
          "market": "btcusdt",
          "price": 10111,
          "taker_type": "sell",
          "total": 6102.9996,
        },
        Object {
          "amount": 0.09,
          "created_at": 1599462444,
          "id": 502501197,
          "market": "btcusdt",
          "price": 10115,
          "taker_type": "buy",
          "total": 910.35,
        },
        Object {
          "amount": 0.0319,
          "created_at": 1599462416,
          "id": 502501196,
          "market": "btcusdt",
          "price": 10114,
          "taker_type": "sell",
          "total": 322.6366,
        },
        Object {
          "amount": 0.0097,
          "created_at": 1599462404,
          "id": 502501195,
          "market": "btcusdt",
          "price": 10115,
          "taker_type": "sell",
          "total": 98.1155,
        },
        Object {
          "amount": 0.9358,
          "created_at": 1599462362,
          "id": 502501192,
          "market": "btcusdt",
          "price": 10103,
          "taker_type": "sell",
          "total": 9454.3874,
        },
        Object {
          "amount": 0.0097,
          "created_at": 1599462290,
          "id": 502501188,
          "market": "btcusdt",
          "price": 10096,
          "taker_type": "sell",
          "total": 97.9312,
        },
        Object {
          "amount": 0.93657365,
          "created_at": 1599462259,
          "id": 502501186,
          "market": "btcusdt",
          "price": 10095,
          "taker_type": "sell",
          "total": 9454.71099675,
        },
        Object {
          "amount": 0.06242635,
          "created_at": 1599462259,
          "id": 502501187,
          "market": "btcusdt",
          "price": 10094,
          "taker_type": "sell",
          "total": 630.1315769,
        },
        Object {
          "amount": 0.974,
          "created_at": 1599462227,
          "id": 502501185,
          "market": "btcusdt",
          "price": 10097,
          "taker_type": "sell",
          "total": 9834.478,
        },
        Object {
          "amount": 0.3348,
          "created_at": 1599462176,
          "id": 502501182,
          "market": "btcusdt",
          "price": 10090,
          "taker_type": "sell",
          "total": 3378.132,
        },
        Object {
          "amount": 0.00026819,
          "created_at": 1599462096,
          "id": 502501174,
          "market": "btcusdt",
          "price": 10082,
          "taker_type": "buy",
          "total": 2.70389158,
        },
        Object {
          "amount": 0.00973181,
          "created_at": 1599462096,
          "id": 502501173,
          "market": "btcusdt",
          "price": 10081,
          "taker_type": "buy",
          "total": 98.10637661,
        },
        Object {
          "amount": 0.00000797,
          "created_at": 1599462075,
          "id": 502445170,
          "market": "btcusdt",
          "price": 10061,
          "taker_type": "buy",
          "total": 0.08018617,
        },
        Object {
          "amount": 0.00724,
          "created_at": 1599462075,
          "id": 502445169,
          "market": "btcusdt",
          "price": 10061,
          "taker_type": "buy",
          "total": 72.84164,
        },
        Object {
          "amount": 0.005,
          "created_at": 1599462071,
          "id": 502445168,
          "market": "btcusdt",
          "price": 10061,
          "taker_type": "buy",
          "total": 50.305,
        },
        Object {
          "amount": 0.01289261,
          "created_at": 1599462070,
          "id": 502445167,
          "market": "btcusdt",
          "price": 10061,
          "taker_type": "buy",
          "total": 129.71254921,
        },
        Object {
          "amount": 0.02646376,
          "created_at": 1599462070,
          "id": 502445166,
          "market": "btcusdt",
          "price": 10061,
          "taker_type": "buy",
          "total": 266.25188936,
        },
        Object {
          "amount": 0.0463,
          "created_at": 1599462054,
          "id": 502445154,
          "market": "btcusdt",
          "price": 10065,
          "taker_type": "sell",
          "total": 466.0095,
        },
        Object {
          "amount": 0.00000416,
          "created_at": 1599462054,
          "id": 502445152,
          "market": "btcusdt",
          "price": 10067,
          "taker_type": "sell",
          "total": 0.04187872,
        },
        Object {
          "amount": 0.0097,
          "created_at": 1599462054,
          "id": 502445153,
          "market": "btcusdt",
          "price": 10065,
          "taker_type": "sell",
          "total": 97.6305,
        },
        Object {
          "amount": 0.15341,
          "created_at": 1599462054,
          "id": 502445151,
          "market": "btcusdt",
          "price": 10067,
          "taker_type": "sell",
          "total": 1544.37847,
        },
        Object {
          "amount": 0.30055453,
          "created_at": 1599462053,
          "id": 502445150,
          "market": "btcusdt",
          "price": 10067,
          "taker_type": "sell",
          "total": 3025.68245351,
        },
        Object {
          "amount": 0.10264684,
          "created_at": 1599462052,
          "id": 502445149,
          "market": "btcusdt",
          "price": 10067,
          "taker_type": "sell",
          "total": 1033.34573828,
        },
        Object {
          "amount": 0.00271947,
          "created_at": 1599462051,
          "id": 502445148,
          "market": "btcusdt",
          "price": 10067,
          "taker_type": "sell",
          "total": 27.37690449,
        },
        Object {
          "amount": 0.005,
          "created_at": 1599461937,
          "id": 502445142,
          "market": "btcusdt",
          "price": 10071,
          "taker_type": "buy",
          "total": 50.355,
        },
        Object {
          "amount": 0.29795,
          "created_at": 1599461929,
          "id": 502445138,
          "market": "btcusdt",
          "price": 10069,
          "taker_type": "sell",
          "total": 3000.05855,
        },
      ]
    }
  />
</Component>
`;
